diff -Nur aircrack-ng-1.2-rc2/scripts/airmon-ng aircrack-ng-1.2-rc2-patched/scripts/airmon-ng
--- aircrack-ng-1.2-rc2/scripts/airmon-ng	2015-04-09 06:05:24.000000000 +0200
+++ aircrack-ng-1.2-rc2-patched/scripts/airmon-ng	2015-05-07 14:04:47.195184565 +0200
@@ -100,18 +100,19 @@
 	exit 1
 fi
 
-if [ -d /proc/bus/pci ]
-then
-	if [ ! -x "$(command -v lspci 2>&1)" ]
-	then
-		printf "Please install lspci from your distro's package manager.\n"
-		exit 1
-	else
-		LSPCI=1
-	fi
-else
-	LSPCI=0
-fi
+LSPCI=0
+#if [ -d /proc/bus/pci ]
+#then
+#	if [ ! -x "$(command -v lspci 2>&1)" ]
+#	then
+#		printf "Please install lspci from your distro's package manager.\n"
+#		exit 1
+#	else
+#		LSPCI=1
+#	fi
+#else
+#	LSPCI=0
+#fi
 
 if [ -f /proc/modules ]
 then
@@ -1189,7 +1190,10 @@
 		else
 			ifacet="\t\t"
 		fi
-		printf "${PHYDEV}\t${iface}${ifacet}${DRIVER}${DRIVERt}${CHIPSET}"
+		if [ "$PHYDEV" != "null" ]
+		then
+			printf "${PHYDEV}\t${iface}${ifacet}${DRIVER}${DRIVERt}${CHIPSET}"
+		fi
 	fi
 
 	if [ x$MAC80211 = "x1" ]
