--- a/scripts/airmon-ng.linux
+++ b/scripts/airmon-ng.linux
@@ -105,27 +105,28 @@
 	LSUSB=0
 fi
 
-if [ -d /sys/bus/pci ] || [ -d /sys/bus/pci_express ] || [ -d /proc/bus/pci ]; then
-	PCI_DEVICES=0
-	if [ -d /sys/bus/pci/devices ] && [ "$(ls -1 /sys/bus/pci/devices 2>/dev/null | wc -l)" != '0' ]; then
-		PCI_DEVICES=1
-	elif [ -r /proc/bus/pci/devices ] && [ -n "$(cat /proc/bus/pci/devices 2>/dev/null)" ]; then
-		PCI_DEVICES=1
-	elif [ -d /sys/bus/pci_express/devices ] && [ -n "$(ls -1 /sys/bus/pci_express/devices 2>/dev/null | wc -l)" != '0' ]; then
-		PCI_DEVICES=1
-	fi
+LSPCI=0
+# if [ -d /sys/bus/pci ] || [ -d /sys/bus/pci_express ] || [ -d /proc/bus/pci ]; then
+# 	PCI_DEVICES=0
+# 	if [ -d /sys/bus/pci/devices ] && [ "$(ls -1 /sys/bus/pci/devices 2>/dev/null | wc -l)" != '0' ]; then
+# 		PCI_DEVICES=1
+# 	elif [ -r /proc/bus/pci/devices ] && [ -n "$(cat /proc/bus/pci/devices 2>/dev/null)" ]; then
+# 		PCI_DEVICES=1
+# 	elif [ -d /sys/bus/pci_express/devices ] && [ -n "$(ls -1 /sys/bus/pci_express/devices 2>/dev/null | wc -l)" != '0' ]; then
+# 		PCI_DEVICES=1
+# 	fi
 
-	if [ ${PCI_DEVICES} -eq 0 ]; then
-		LSPCI=0
-	elif [ ! -x "$(command -v lspci 2>&1)" ]; then
-		printf "Please install lspci from your distro's package manager.\n"
-		exit 1
-	else
-		LSPCI=1
-	fi
-else
-	LSPCI=0
-fi
+# 	if [ ${PCI_DEVICES} -eq 0 ]; then
+# 		LSPCI=0
+# 	elif [ ! -x "$(command -v lspci 2>&1)" ]; then
+# 		printf "Please install lspci from your distro's package manager.\n"
+# 		exit 1
+# 	else
+# 		LSPCI=1
+# 	fi
+# else
+# 	LSPCI=0
+# fi
 
 if [ -f /proc/modules ] || [ -d /sys/module ]; then
 	if [ ! -x "$(command -v modprobe 2>&1)" ]; then
