--- a/scripts/airmon-ng.linux
+++ b/scripts/airmon-ng.linux
@@ -105,16 +105,17 @@ else
 	LSUSB=0
 fi
 
-if [ -d /sys/bus/pci ] || [ -d /sys/bus/pci_express ] || [ -d /proc/bus/pci ]; then
-	if [ ! -x "$(command -v lspci 2>&1)" ]; then
-		printf "Please install lspci from your distro's package manager.\n"
-		exit 1
-	else
-		LSPCI=1
-	fi
-else
-	LSPCI=0
-fi
+LSPCI=0
+#if [ -d /sys/bus/pci ] || [ -d /sys/bus/pci_express ] || [ -d /proc/bus/pci ]; then
+#	if [ ! -x "$(command -v lspci 2>&1)" ]; then
+#		printf "Please install lspci from your distro's package manager.\n"
+#		exit 1
+#	else
+#		LSPCI=1
+#	fi
+#else
+#	LSPCI=0
+#fi
 
 if [ -f /proc/modules ] || [ -d /sys/module ]; then
 	if [ ! -x "$(command -v modprobe 2>&1)" ]; then
